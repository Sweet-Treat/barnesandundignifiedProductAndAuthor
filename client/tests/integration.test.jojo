/*import 'jsdom-global/register'; //at the top of file , even  , before importing react
//import regeneratorRuntime from "regenerator-runtime";
import App from '../src/components/app.jsx';
import ProductDetails from '../src/components/productDetails.jsx';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

//import nock from 'nock' // v9.0.2
//const host = 'http://localhost:5001';


configure({ adapter: new Adapter() });

import { shallow, mount, render } from 'enzyme';

describe('Checking that App comoponent really use the express.js server', () => {

  it ('renders the inner productDetails',  async () => {
    const wrapper =  await mount(<App isbn13={'9781524763169'} />)
    await wrapper.instance().componentDidMount();

    expect(wrapper.debug()).toMatchSnapshot();
    expect(wrapper.find(ProductDetails).length).toEqual(1);
  });
});*/






