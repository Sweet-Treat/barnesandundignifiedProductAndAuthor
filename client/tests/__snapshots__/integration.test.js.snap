// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App react test renders the inner productDetails 1`] = `
"<App isbn13=\\"9781524763169\\">
  <div>
    <h2>
      Product Details
    </h2>
    <ProductDetails record={{...}}>
      <table className=\\"productDetailsTB\\">
        <tbody>
          <tr>
            <th>
              ISBN-13: 
            </th>
            <td />
          </tr>
          <tr>
            <th>
              Publisher: 
            </th>
            <td>
              <a href=\\"\\" />
            </td>
          </tr>
          <tr>
            <th>
              Publication Date: 
            </th>
            <td>
              Invalid Date
            </td>
          </tr>
          <tr>
            <th>
              Pages: 
            </th>
            <td />
          </tr>
          <tr>
            <th>
              Sales Rank: 
            </th>
            <td />
          </tr>
        </tbody>
      </table>
    </ProductDetails>
  </div>
</App>"
`;

exports[`Checking that App comoponent really use the express.js server renders the inner productDetails 1`] = `
"<App isbn13=\\"9781524763169\\">
  <div>
    <h2>
      Product Details
    </h2>
    <ProductDetails record={{...}}>
      <table className=\\"productDetailsTB\\">
        <tbody>
          <tr>
            <th>
              ISBN-13: 
            </th>
            <td />
          </tr>
          <tr>
            <th>
              Publisher: 
            </th>
            <td>
              <a href=\\"\\" />
            </td>
          </tr>
          <tr>
            <th>
              Publication Date: 
            </th>
            <td>
              Invalid Date
            </td>
          </tr>
          <tr>
            <th>
              Pages: 
            </th>
            <td />
          </tr>
          <tr>
            <th>
              Sales Rank: 
            </th>
            <td />
          </tr>
        </tbody>
      </table>
    </ProductDetails>
  </div>
</App>"
`;
